### PR TITLE
Add Android CI

### PR DESCRIPTION
**Pull request by Aang23**
This PR adds CI compilation checks on Android NDK, for all 4 supported and required architectures.

Of course, armeabi-v7a currently fails, as I didn't think addressing this issue was to be done in this PR.

Initially mentioned in https://github.com/gnuradio/volk/issues/589

### CODE REVIEW

**Comment by jdemel** [comment]
Thanks for the PR. It's nice to see that things compile in most cases. I'm still curious. Would it be possible to run our tests as well? I assume that'd require smth like our `aarch64` test approach.

**Comment by Aang23** [comment]
It should be possible with a bit of work - gonna see if I can figure it out. But at the same time, clang and the same architectures are already being tested on Linux, so it is highly unlikely it would pass there and break on Android.

**Comment by jdemel** [comment]
Well, #589 came up because of an Android specific issue.
Also, out of these for builds `armeabi-v7a`, `arm64-v8a`, `x86`, `x86_64`, we only test `arm64-v8a` and `x86_64` thoroughly. The other two are neglected.

File: .github/workflows/android_build.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+jobs:
+  build:
+    name: Build on Android NDK ${{ matrix.arch.name }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: 
+          - { name: armeabi-v7a }
```
Comment (jdemel) [inline .github/workflows/android_build.yml]: As long as this test fails, I suggest to add smth like `allow-failure`. Maybe `continue-on-error` is the GH Action equivalent of the TravisCI `allow-failure` option.

**Comment by Aang23** [comment]
Good point, though it was a compile-time issue.
I've done some (relatively) quick research and testing, and it does not appear to be easy at all to run tests without running some sort of Android emulator / device. Maybe someone else will have an idea at some point. You're not really supposed to run actual executable on Android after all...

I fixed up another another issue I found when building static libs.

**Review by jdemel — APPROVED** [review]
LGTM. This PR would be a nice addition to test VOLK on more platforms.

**Review by michaelld — APPROVED** [review]
LGTM ; thx!

**Pull request merged**